src/component/user/Authentication/Authentication.test.tsx

```diff
@@ -0,0 +1,135 @@
+import Authentication from 'component/user/Authentication/Authentication';
+import { render, screen } from '@testing-library/react';
+import {
+    LOGIN_PASSWORD_ID,
+    LOGIN_EMAIL_ID,
+    LOGIN_BUTTON,
+    AUTH_PAGE_ID,
+    SSO_LOGIN_BUTTON,
+} from 'utils/testIds';
+import React from 'react';
+import { TestContext } from 'utils/testContext';
+import { testServerSetup, testServerRoute } from 'utils/testServer';
+
+const server = testServerSetup();
+
+test('should render password auth', async () => {
+    testServerRoute(server, '*', {
+        defaultHidden: false,
+        message: 'You must sign in in order to use Unleash',
+        path: '/auth/simple/login',
+        type: 'password',
+        options: [],
+    });
+
+    render(
+        <TestContext>
+            <Authentication redirect="/" />
+        </TestContext>
+    );
+
+    await screen.findByTestId(AUTH_PAGE_ID);
+    await screen.findByTestId(LOGIN_PASSWORD_ID);
+    await screen.findByTestId(LOGIN_EMAIL_ID);
+    await screen.findByTestId(LOGIN_BUTTON);
+});
+
+test('should not render password auth if defaultHidden is true', async () => {
+    testServerRoute(server, '*', {
+        defaultHidden: true,
+        message: 'You must sign in in order to use Unleash',
+        path: '/auth/simple/login',
+        type: 'password',
+        options: [],
+    });
+
+    render(
+        <TestContext>
+            <Authentication redirect="/" />
+        </TestContext>
+    );
+
+    await screen.findByTestId(AUTH_PAGE_ID);
+    expect(screen.queryByTestId(LOGIN_EMAIL_ID)).toBeNull();
+    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
+    expect(screen.queryByTestId(LOGIN_BUTTON)).toBeNull();
+});
+
+test('should render demo auth', async () => {
+    testServerRoute(server, '*', {
+        defaultHidden: false,
+        message: 'You must sign in in order to use Unleash',
+        path: '/auth/demo/login',
+        type: 'demo',
+        options: [],
+    });
+
+    render(
+        <TestContext>
+            <Authentication redirect="/" />
+        </TestContext>
+    );
+
+    await screen.findByTestId(AUTH_PAGE_ID);
+    expect(screen.getByTestId(LOGIN_EMAIL_ID)).not.toBeNull();
+    expect(screen.queryByTestId(LOGIN_PASSWORD_ID)).toBeNull();
+    expect(screen.getByTestId(LOGIN_BUTTON)).not.toBeNull();
+});
+
+test('should render email auth', async () => {
+    testServerRoute(server, '*', {
+        defaultHidden: false,
+        message: 'You must sign in in order to use Unleash',
+        path: '/auth/unsecure/login',
+        type: 'unsecure',
+        options: [],
+    });
+
+    render(
+        <TestContext>
+            <Authentication redirect="/" />
+        </TestContext>
+    );
+
+    await screen.findByTestId(AUTH_PAGE_ID);
+    expect(screen.getByTestId(LOGIN_EMAIL_ID)).not.toBeNull();
```

Comment:
maybe `toBeInTheDocument()` or something similar? Jest + Testing library have a lot of helpers/sugar that make reading tests more natural